File: code-samples/messaging/send-fax.java
package Send_Fax;

import java.io.IOException;

import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class Send_Fax {
    static String RECIPIENT = System.getenv("FAX_RECIPIENT");
    static RestClient restClient;
  
    public static void main(String[] args) {
        var obj = new Send_Fax();
	restClient = new RestClient( System.getenv("RC_CLIENT_ID"),
				     System.getenv("RC_CLIENT_SECRET"),
				     System.getenv("RC_SERVER_URL") );
	try {
	    restClient.authorize( System.getenv("RC_JWT") );
            obj.sendFax();
	} catch (RestException | IOException e) {
	    e.printStackTrace();
	}
    }
    
    public static void sendFax() throws RestException, IOException{
        CreateFaxMessageRequest postParameters = new CreateFaxMessageRequest();
        postParameters.to = new MessageStoreCallerInfoRequest[]{new MessageStoreCallerInfoRequest().phoneNumber(RECIPIENT)};
        postParameters.faxResolution = "High";
        postParameters.coverPageText = "This is a demo Fax page from Java";
        Attachment attachment = new Attachment();
        attachment.fileName = "test.jpg";
        attachment.contentType = "image/jpeg";
        attachment.bytes = Files.readAllBytes(Paths.get("./src/test/resources/test.jpg"));
        Attachment[] attachments = new Attachment[] { attachment };
        postParameters.attachments = attachments;
        
        var response = rc.restapi().account().extension().fax().post(postParameters);
        System.out.println("Fax sent. Delivery status: " + response.messageStatus);
    }
}

File: code-samples/messaging/send-fax.cs
using System;
using System.Threading.Tasks;
using RingCentral;

namespace Send_Fax
{
    class Program
    {
        static RestClient restClient;
        static void Main(string[] args)
        {
            restClient = new RestClient(
                Environment.GetEnvironmentVariable("RC_CLIENT_ID"),
                Environment.GetEnvironmentVariable("RC_CLIENT_SECRET"),
                Environment.GetEnvironmentVariable("RC_SERVER_URL"));
            restClient.Authorize(
                Environment.GetEnvironmentVariable("RC_JWT")).Wait();
            send_fax().Wait();
        }
        
        static private async Task send_fax()
        {
            var requestParams = new CreateFaxMessageRequest();
            var attachment = new Attachment { fileName = "test.jpg", contentType = "image/jpeg", bytes = System.IO.File.ReadAllBytes("test.jpg") };
            var attachments = new Attachment[] { attachment };
            requestParams.attachments = attachments;
            requestParams.to = new MessageStoreCalleeInfoRequest[] { new MessageStoreCalleeInfoRequest { phoneNumber = RECIPIENT } };
            requestParams.faxResolution = "High";
            requestParams.coverPageText = "This is a demo Fax page from C#";
            var resp = await rc.Restapi().Account().Extension().Fax().Post(requestParams);
            Console.WriteLine("Fax sent. Message status: " + resp.messageStatus);
        }
    }
}

